keps/1282-pods-ready-requeue-strategy/README.md

```diff
@@ -208,39 +209,37 @@ type RequeueState struct {
 }
 ```
 
-Add a new field `evictionStats` to the WorkloadStatus struct to track the number of evictions by type and reason:
+Add a new field, `evictionState`, to the WorkloadStatus struct to track the number of evictions by reason:
 
 ```go
 type WorkloadStatus struct {
-	// evictionStats tracks eviction statistics by type and reason.
+    // evictionState tracks eviction statistics by reason.
     //
     // +optional
     // +listType=map
-    // +listMapKey=type
+    // +listMapKey=reason
     // +patchStrategy=merge
-    // +patchMergeKey=type
-    EvictionStats []Eviction `json:"evictionStats,omitempty"`
+    // +patchMergeKey=reason
+    EvictionStates []WorkloadEvictionState `json:"evictionStates,omitempty"`
 }
 
-type Eviction struct {
-    // type identifies the eviction type.
+type WorkloadEvictionState struct {
+    // reason specifies the programmatic identifier for the eviction cause.
     //
     // +required
     // +kubebuilder:validation:Required
-    // +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
     // +kubebuilder:validation:MaxLength=316
-    Type string `json:"type"`
-    
-    // reason specifies the programmatic identifier for the eviction cause.
+    Reason string `json:"reason"`
+
+    // detailedReason specifies a finer-grained explanation that complements the eviction cause.
+    // This may be an empty string.
     //
     // +required
     // +kubebuilder:validation:Required
-    // +kubebuilder:validation:MaxLength=1024
-    // +kubebuilder:validation:MinLength=1
-    // +kubebuilder:validation:Pattern=`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`
-    Reason string `json:"reason"`
-    
-    // count tracks the number of evictions for this type and reason.
+    // +kubebuilder:validation:MaxLength=316
+    DetailedReason string `json:"detailedReason"`
+
+    // count tracks the number of evictions for this reason and detailed reason.
     //
```

Comment:
I guess we will need LastUpdate soon. Anyway, this can wait until we have a PR that uses it.